src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9773,11 +9773,27 @@ bool shouldTreatAsModifier()
                 using var _ = this.GetDisposableResetPoint(resetOnDispose: true);
 
                 Debug.Assert(this.CurrentToken.Kind == SyntaxKind.IdentifierToken);
-                this.EatToken();
 
-                return IsDeclarationModifier(this.CurrentToken.Kind) ||
-                    IsAdditionalLocalFunctionModifier(this.CurrentToken.Kind) ||
-                    (ScanType() != ScanTypeFlags.NotType && this.CurrentToken.Kind == SyntaxKind.IdentifierToken);
+                do
+                {
+                    this.EatToken();
+
+                    using var _2 = this.GetDisposableResetPoint(resetOnDispose: true);
```

Comment:
>using var _2 = this.GetDisposableResetPoint(resetOnDispose: true); [](http://example.com/codeflow?start=20&length=66)

Consider refactoring the code so that this reset point is allocated only when ScanType is actually called.

Reply:
Done, but just interesting: what allocation is done here? `DisposableResetPoint`, returned by the `GetDisposableResetPoint` is a `ref struct`, so it **must not** allocate, right?

Reply:
Stack space is also a resource that can have impact on parsing. It's cheap, but does consume a constrained resource.  Here is likely not a major issue, but general hygiene in these areas is appreciated. Thanks!